src/lib.rs

```diff
@@ -845,7 +847,8 @@ impl Node {
 	///
 	/// Returns a temporary channel id
 	pub fn connect_open_channel(
-		&self, node_pubkey_and_address: &str, channel_amount_sats: u64, announce_channel: bool,
+		&self, node_id: &PublicKey, address: &SocketAddr, channel_amount_sats: u64,
```

Comment:
Hmm... I'd rather take by value instead of cloning if the method needs to own the data. I believe that is more idiomatic. The caller can clone if need be.